packages/devtools_app/lib/src/memory/memory_controller.dart

```diff
@@ -583,27 +626,76 @@ class MemoryController extends DisposableController
         ?.getHeapSnapshotGraph(serviceManager?.isolateManager?.selectedIsolate);
   }
 
+  final _monitorAllocationsUpdated = ValueNotifier<int>(0);
+
+  /// Last column sorted and sort direction in allocation monitoring. As table
+  /// is reconstructed e.g., reset, etc. remembers user's sorting preference.
+  ColumnData<ClassHeapDetailStats> sortedMonitorColumn;
+  SortDirection sortedMonitorDirection;
+
+  ValueListenable<int> get monitorAllocationsChanged =>
+      _monitorAllocationsUpdated;
+
+  DateTime monitorTimestamp;
+
+  var _monitorAllocations = <ClassHeapDetailStats>[];
+
+  List<ClassHeapDetailStats> get monitorAllocations => _monitorAllocations;
+
+  set monitorAllocations(List<ClassHeapDetailStats> allocations) {
+    _monitorAllocations = allocations;
+    // Clearing allocations reset ValueNotifier to zero.
+    allocations.isNotEmpty
+        ? _monitorAllocationsUpdated.value++
+        : _monitorAllocationsUpdated.value = 0;
+  }
+
   Future<List<ClassHeapDetailStats>> resetAllocationProfile() =>
       getAllocationProfile(reset: true);
 
-  // 'reset': true to reset the object allocation accumulators
+  /// 'reset': true to reset the object allocation accumulators
   Future<List<ClassHeapDetailStats>> getAllocationProfile({
-    bool reset = false,
+    reset = false,
   }) async {
+    if (!await isIsolateLive(_isolateId)) return [];
+
     AllocationProfile allocationProfile;
-    try {
-      allocationProfile = await serviceManager.service.getAllocationProfile(
-        _isolateId,
-        reset: reset,
-      );
-    } on SentinelException catch (_) {
-      return [];
+    allocationProfile = await serviceManager.service.getAllocationProfile(
+      _isolateId,
+      reset: reset,
+    );
+
+    final lastReset = allocationProfile.dateLastAccumulatorReset;
+    if (lastReset != null) {
+      final resetTimestamp = DateTime.fromMillisecondsSinceEpoch(lastReset);
+      debugLogger('Last Allocation Reset @ '
+          '${MemoryController.formattedTimestamp(resetTimestamp)}');
     }
-    return allocationProfile.members
+
+    final allocations = allocationProfile.members
         .map((ClassHeapStats stats) => ClassHeapDetailStats(stats.json))
         .where((ClassHeapDetailStats stats) {
       return stats.instancesCurrent > 0 || stats.instancesAccumulated > 0;
     }).toList();
+
+    return allocations;
+  }
+
+  Future<void> computeLibraries() async {
+    // TODO(terry): Review why unawaited is necessary.
+    unawaited(serviceManager.service.getVM().then((vm) {
```

Comment:
this unawaited seems wrong. The Future returned by the method doesn't match when the libraries are actually computed. On the surface it seems like you could write
```dart
final vm = await serviceManager.service.getVM();
final isolates = await Future.wait(vm.isolates.map((IsolateRef ref) => serviceManager.service.getIsolate(ref.id))
...

```

Reply:
Removed function no longer used.